feature_engine/sanity_check/similar_columns.py

```diff
@@ -0,0 +1,125 @@
+from typing import Any
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from feature_engine.validation import _return_tags
+
+from feature_engine.dataframe_checks import (
+    _check_contains_inf,
+    _check_contains_na,
+    _is_dataframe,
+)
+from feature_engine.variable_manipulation import _find_all_variables
+
+
+class SimilarColumns(BaseEstimator, TransformerMixin):
+    """Ensure that similar columns are in test and train dataset.
+
+    Parameters
+    ----
+    None
+
+    """
+
+    def __init__(
+        self,
+        fill_value: Any = np.nan,
+        missing_values: str = "raise",
+        verbose: bool = True
+    ):
+
+        if missing_values not in ["raise", "ignore"]:
+            raise ValueError("missing_values takes only values 'raise' or 'ignore'.")
+
+        if not isinstance(verbose, bool):
+            raise ValueError("verbose takes only booleans True and False")
+
+        self.fill_value = fill_value
+        self.missing_values = missing_values
+        self.verbose = verbose
+
+    def _check_input(self, X: pd.DataFrame) -> pd.DataFrame:
+        X = _is_dataframe(X)
+
+        if self.missing_values == "raise":
+            # check if dataset contains na
+            _check_contains_na(X, self.variables_)
+            _check_contains_inf(X, self.variables_)
+
+        return X
+
+    def fit(self, X: pd.DataFrame, y: pd.Series = None):
+        """Fit columns schema
+
+        Parameters
+        ----------
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The input dataframe
+
+        y: None
+            y is not needed for this transformer. You can pass y or None.
+
+        """
+        X = _is_dataframe(X)
+
+        self.n_features_in_ = X.shape[1]
+
+        self.variables_ = _find_all_variables(X, list(X.columns))
+
+        X = self._check_input(X)
```

Comment:
this should be at the top of fit() instead of line 67.

Reply:
We cannot do that, because self._check_input(X) need self._variables to be define.